deepchecks/nlp/utils/text_data_plot.py

```diff
@@ -0,0 +1,302 @@
+          sequence of sequences, with the sequence for each sample being a binary vector, representing the presence of
+          the i-th label in that sample.
+        - token_classification label - For token classification the accepted label format is the IOB format or similar
+          to it. The Label must be a sequence of sequences of strings or integers, with each sequence corresponding to
+          a sample in the tokenized text, and exactly the length of the corresponding tokenized text.
+
+    Returns
+    -------
+    Plotly Figure instance.
+    """
+    specs = [[{'type': 'pie'}, {'type': 'table'}] if label is not None else [{'type': 'table', 'colspan': 2}, None]] + \
+        [[{'type': 'xy', 'colspan': 2}, None] for _ in range(len(properties.columns))]
+
+    subplot_titles = []
+    if label is not None:
+        annotated_samples = pd.notna(label).sum()
+        subplot_titles.append(f'Label Distribution<br><sup>Out of {annotated_samples} annotated samples</sup><br><br>')
+
+    subplot_titles.append('')  # Empty title for table figure
+    if not properties.empty:
+        for prop_name in properties:
+            if prop_name in TEXT_PROPERTIES_DESCRIPTION:
+                subplot_titles.append(f'{prop_name} Property Distribution<sup><a href="{get_docs_link()}nlp/'
+                                      'usage_guides/nlp_properties.html#deepchecks-built-in-properties">&#x24D8;</a>'
+                                      f'</sup><br><sup>{TEXT_PROPERTIES_DESCRIPTION[prop_name]}</sup>')
+
+    fig = make_subplots(rows=len(properties.columns) + 1, cols=2, specs=specs, subplot_titles=subplot_titles,
+                        row_heights=[1.5] + [1.0] * len(properties.columns))
+
+    # Create label distribution if label is provided
+    if label is not None:
+        if is_multi_label:
+            df_label = pd.DataFrame(label).fillna(0)
+            label_counts = pd.Series(np.sum(df_label.to_numpy(), axis=0))
+        elif task_type == TaskType.TOKEN_CLASSIFICATION:
+            hashmap = {}
+            for val in label:
+                flattened_array = pd.Series(np.array(val).flatten()).fillna('NaN').to_numpy()
+                unique_values, counts = np.unique(flattened_array, return_counts=True)
+                for label_value, count in zip(unique_values, counts):
+                    if label_value != 'NaN':
+                        hashmap[label_value] = hashmap[label_value] + count if label_value in hashmap else count
+            label_counts = pd.Series(list(hashmap.values()), index=list(hashmap))
+        else:
+            label_counts = pd.Series(label).value_counts()
+
+        label_counts.sort_values(ascending=False, inplace=True)
+        labels_to_display = label_counts[:max_num_labels_to_show]
+        labels_to_display.index = [break_to_lines_and_trim(str(label)) for label in list(labels_to_display.index)]
```

Comment:
continue of the above comment -  use model class argument here

Reply:
Done. And tested by passing a list of class names:
<img width="630" alt="image" src="https://github.com/deepchecks/deepchecks/assets/136261806/c31dcbe1-9e4a-45ac-aaed-4d4dee5c8857">